tests/raftstore/test_stale_peer.rs

```diff
@@ -0,0 +1,179 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A module contains test cases of stale peers gc.
+
+use std::time::Duration;
+use std::thread;
+
+use kvproto::eraftpb::MessageType;
+use kvproto::raft_serverpb::{RegionLocalState, PeerState};
+use tikv::raftstore::store::{keys, Peekable};
+
+use super::cluster::{Cluster, Simulator};
+use super::node::new_node_cluster;
+use super::server::new_server_cluster;
+use super::transport_simulate::{Isolate, IsolateRegionStore};
+use super::util::*;
+
+/// A helper function for testing the behaviour of the gc of stale peer
+/// which is out of region.
+/// If a peer detects the leader is missing for a specified long time,
+/// it should consider itself as a stale peer which is removed from the region.
+/// This test case covers the following scenario:
+/// At first, there are three peer A, B, C in the cluster, and A is leader.
+/// Peer B gets down. And then A adds D, E, F int the cluster.
+/// Peer D becomes leader of the new cluster, and then removes peer A, B, C.
+/// After all these peer in and out, now the cluster has peer D, E, F.
+/// If peer B goes up at this moment, it still thinks it is one of the cluster
+/// and has peers A, C. However, it could not reach A, C since they are removed from
+/// the cluster or probably destroyed.
+/// Meantime, D, E, F would not reach B, Since it's not in the cluster anymore.
+/// In this case, Peer B would notice that the leader is missing for a long time,
+/// and it would check with pd to confirm whether it's still a member of the cluster.
+/// If not, it should destroy itself as a stale peer which is removed out already.
+fn test_stale_peer_out_of_region<T: Simulator>(cluster: &mut Cluster<T>) {
+    // Revise raft base tick interval small to make this test case run fast.
```

Comment:
I think the default configuration should be enough. We have reconfigured these settings in `util::new_store_cfg` and `util::new_server_config`.